augur/filter.py

```diff
@@ -12,17 +12,21 @@
 
 comment_char = '#'
 
-def read_vcf(compressed, input_file):
```

Comment:
probably should not change the API, how "public" is this function?   Is there a process for documenting API changes?

Reply:
I'll defer to the project maintainers, but it doesn't seem like a function that would be useful to the public. There's no public-facing documentation, anyway.

It's actually a bit misnamed, since it doesn't read the entire VCF data structure. It should probably be called something like `read_sample_names_from_vcf`.

Reply:
My instincts on this are tuned to widely used open source projects so I am possibly too cautious. I definitely agree we should defer to the maintainers here.

If there is not a process for documenting API changes, strongly suggest adding one.